openfl/component/director/director.py

```diff
@@ -74,16 +91,21 @@ async def set_new_experiment(
             experiment_archive_path: Path,
     ) -> bool:
         """Set new experiment."""
-        experiment = Experiment(
-            name=experiment_name,
-            archive_path=experiment_archive_path,
-            collaborators=list(collaborator_names),
-            users=[sender_name],
-            sender=sender_name,
-            init_tensor_dict=tensor_dict,
-        )
-        self.experiments_registry.add(experiment)
-        return True
+        open_ports_list = self.settings.get('open_ports', OPEN_PORTS_LIST)
+        open_port = self.get_open_port(open_ports_list, sender_name)
+        if open_port:
+            experiment = Experiment(
+                name=experiment_name,
+                archive_path=experiment_archive_path,
+                collaborators=list(collaborator_names),
+                users=[sender_name],
+                sender=sender_name,
+                init_tensor_dict=tensor_dict,
+                agg_port=open_port,
+            )
+            self.experiments_registry.add(experiment)
+            return True
+        logger.info(f'Could not find open ports out of the list: {open_ports_list}')
```

Comment:
This should be changed. In the case that all ports are used, this would result in the user not being able to submit their experiment.